src/VirtualPage.php

```diff
@@ -40,26 +37,18 @@ protected function init(): void
     /**
      * Set callback function of virtual page.
      *
-     * Note that only one callback function can be defined.
-     *
-     * @param array $fx   Need this to be defined as array otherwise we get warning in PHP7
-     * @param mixed $junk
+     * @param callable $fx   Need this to be defined as array otherwise we get warning in PHP7
+     * @param mixed    $junk
      *
      * @return $this
      */
-    public function set($fx = [], $junk = null)
+    public function set($fx = null, $junk = null)
     {
-        if (!$fx) {
-            return $this;
+        if (!$fx || !is_callable($fx)) {
+            throw new Exception('Virtual page required a callable.');
```

Comment:
Typo here, "requires", not past tense "required" ;)

Reply:
update - thanks